sql/log.cc

```diff
@@ -3048,7 +3069,7 @@ bool MYSQL_LOG::open(
   DBUG_RETURN(0);
 
 err:
-  sql_print_error(fatal_log_error, name, errno);
```

Comment:
MySQL had an additional `log_type == LOG_BIN` requirement for this condition:
```c++
  if (log_type == LOG_BIN && binlogging_impossible_mode == ABORT_SERVER)
  {
    …
    my_error(ER_BINLOG_LOGGING_IMPOSSIBLE …
```
Why is it there?

Reply:
It ensures we only apply the error logic to the binary log, and not other logs. That is, the binlog (of type `MYSQL_BIN_LOG`) is derived from `MYSQL_LOG`, and this is the general `MYSQL_LOG::open` function.